### PR TITLE
Fix stacking when setting option (#3172)

### PR DESCRIPTION
**Pull request by yotamberk**
solves #3172

### CODE REVIEW

File: misc/RELEASE_CHECKLIST_TEMPLATE.md [hunk 1]
```diff
@@ -80,4 +80,4 @@ If we would merge the development branch would overwrite this. To solve this we
 - [ ] FORCE-Push the branches to github: `git push --force && git push --tag`
 - [ ] [Re-Enable branch protection](https://github.com/almende/vis/settings/branches/develop) (enable ALL checkboxes) for `develop`.
 
-DONE!
+DONE!
```
Comment (bradh) [inline misc/RELEASE_CHECKLIST_TEMPLATE.md]: Did you want to include this part?

File: lib/timeline/component/ItemSet.js [hunk 2]
```diff
@@ -634,9 +634,14 @@ ItemSet.prototype.redraw = function() {
   var visibleInterval = range.end - range.start;
   var zoomed = (visibleInterval != this.lastVisibleInterval) || (this.props.width != this.props.lastWidth);
   var scrolled = range.start != this.lastRangeStart;
-  var forceRestack = (zoomed || scrolled);
+  var changedStack = options.stack != this.lastStack
+  var changedStackSubgroups = options.stackSubgroups != this.lastStackSubgroups
+  var forceRestack = (zoomed || scrolled || changedStack || changedStackSubgroups);
   this.lastVisibleInterval = visibleInterval;
   this.lastRangeStart = range.start;
+  this.lastStack = options.stack;
```
Comment (bradh) [inline lib/timeline/component/ItemSet.js]: The logic is good, but the `lastStack` would be better as `lastStackOptions` - its not really the "last stacking"

File: lib/timeline/component/ItemSet.js [hunk 3]
```diff
@@ -634,9 +634,14 @@ ItemSet.prototype.redraw = function() {
...
+  this.lastStackSubgroups = options.stackSubgroups;
```
Comment (bradh) [inline lib/timeline/component/ItemSet.js]: Same here - perhaps `lastStackSubGroupOptions` or something like that.

File: misc/RELEASE_CHECKLIST_TEMPLATE.md (hunk 1 above)
Comment (yotamberk) [inline misc/RELEASE_CHECKLIST_TEMPLATE.md]: Not sure what changed here. It doesn't seem like a change

**Pull request merged**